zebra-state/src/service.rs

```diff
@@ -1507,14 +1508,29 @@ impl Service<ReadRequest> for ReadStateService {
 
                 tokio::task::spawn_blocking(move || {
                     span.in_scope(move || {
+                        let end_index = limit
+                            .and_then(|limit| start_index.0.checked_add(limit.0))
```

Comment:
There's test coverage for using the `read::{sapling, orchard}_subtree` methods with a `RangeFrom` as would be used here when this overflows.

Reply:
I was asking if the overflow case of this statement ever runs in our tests. Do our RPC integration tests have test cases that call it?

As far as I remember they mainly use indexes and limits 1 and 2. But it would be quick to add.

Reply:
> Do our RPC integration tests have test cases that call it?

There are snapshot tests for `z_get_subtrees_by_index()` but they use a mock state service with low indexes/limits.

Reply:
I am happy to call this test optional and resolve this comment, because it's an edge case that seems correct from the code.

Are you ok with that, or is the test important enough to spend time on?

Reply:
> Are you ok with that, or is the test important enough to spend time on?

I'm happy with that :)